include/circt/Dialect/FIRRTL/FIRRTLTypes.td

```diff
@@ -67,6 +67,10 @@ def PassiveType : DialectType<FIRRTLDialect,
   CPred<"$_self.isa<FIRRTLType>() && $_self.cast<FIRRTLType>().isPassive()">,
   "a passive type (contain no flips)", "::circt::firrtl::FIRRTLType">;
 
+def RefType : DialectType<FIRRTLDialect,
+   CPred<"$_self.isa<RefType>() && $_self.cast<RefType>().getType().isGround()">,
+  "reference type", "::circt::firrtl::RefType">;
```

Comment:
The ground type restriction is going to be too restrictive eventually. However, I am good to start with this.